server/app/services/geo/esri/EsriClient.java

```diff
@@ -105,6 +105,15 @@ public CompletionStage<AddressSuggestionGroup> getAddressSuggestions(Address add
                         .setState(attributes.get("RegionAbbr").asText())
                         .setZip(attributes.get("Postal").asText())
                         .build();
+                // Suggestion must be a fully formed address, including a number.
+                // Sometimes the esri service returns only a partially formed address.
+                if (candidateAddress.getStreet().isEmpty()
+                    || !candidateAddress.getStreet().matches("^[0-9]+.*")
```

Comment:
Just checking that the street value is not empty should suffice. We risk losing potential valid answers with the regex check.

Reply:
According to https://github.com/civiform/civiform/issues/4443, we also wanted to get rid of suggestions without a number.  I think when you provide a fairly well formed address, sometimes the esri service offers up suggestions that omit the number for some reason.  What kind of addresses might we want to include without a number?  I feel like there are some, I just can't picture it right now. :)

Reply:
I guess P.O. boxes?

Reply:
There could be PO Boxes

Reply:
Hmm....gotta think about how we should handle that.  Maybe it's not so terrible to have a suggestion that doesn't have a number when it should.  Users can just ignore it.

Reply:
Suppose we could try to get fancy and do something like "if a user provided a street address number, remove suggestions that don't have one too" but I fear it might be creating edge cases for ourselves.

Reply:
Yeah I don't like going with a complex plan unless it is causing real problems. I think as long as all the fields have a value it should be fine.

Reply:
👍 Updated